### PR TITLE
Adjust quantizers via segmentation and RDO

### PR DESCRIPTION
**Pull request by YaLTeR**
[AWCY](https://beta.arewecompressedyet.com/?job=master-speed2-914fe8df0df9cdb87bd0f56fa8a57c8a3eab9489%402019-08-01T17%3A03%3A57.470Z&job=temporal-rdo-quantizer-27)

[AWCY but on --speed 5](https://beta.arewecompressedyet.com/?job=master-914fe8df0df9cdb87bd0f56fa8a57c8a3eab9489&job=temporal-rdo-quantizer-26)

With @tdaede we decided I should put this in `--speed 2` as it about doubles the encoding time.

This adds three quantizer index deltas: 0, -15 and 15, and adds searching them to the RDO. This way it's affected by any RDO biasing, be it block importance or activity. This showed better results than me trying to naively set the segment delta according to mean importance.

### CODE REVIEW

File: src/segmentation.rs [hunk 1]
```diff
@@ -10,31 +10,51 @@
...
+  if fi.config.speed_settings.quantizer_rdo {
+    fs.segmentation.enabled = true;
+    fs.segmentation.update_map = true;
+
+    // We don't change the values between frames.
+    fs.segmentation.update_data = fi.primary_ref_frame == PRIMARY_REF_NONE;
 
-    fs.segmentation.features[0][SegLvl::SEG_LVL_ALT_Q as usize] = false;
-    fs.segmentation.data[0][SegLvl::SEG_LVL_ALT_Q as usize] = 0;
+    // A series of AWCY runs with deltas 10, 13, 15, 17, 20 showed this to be
```
Comment (tdaede) [inline src/segmentation.rs]: thanks for commenting this

**Comment by ycho** [comment]
[AWCY result for speed 0](https://beta.arewecompressedyet.com/?job=master_3c79e02_after_segment_s0%402019-08-07T23%3A24%3A28.264Z&job=master_1bc1ba51_before_segment_s0%402019-08-07T23%3A24%3A17.002Z)

**Comment by YaLTeR** [comment]
> [AWCY result for speed 0](https://beta.arewecompressedyet.com/?job=master_3c79e02_after_segment_s0%402019-08-07T23%3A24%3A28.264Z&job=master_1bc1ba51_before_segment_s0%402019-08-07T23%3A24%3A17.002Z)

I think on this link the reverse button should be pressed.

**Comment by ycho** [comment]
> > [AWCY result for speed 0](https://beta.arewecompressedyet.com/?job=master_3c79e02_after_segment_s0%402019-08-07T23%3A24%3A28.264Z&job=master_1bc1ba51_before_segment_s0%402019-08-07T23%3A24%3A17.002Z)
> 
> I think on this link the reverse button should be pressed.

Corrected the order, so no need to press reverse now.

**Pull request merged**